chia/server/server.py

```diff
@@ -611,13 +612,19 @@ async def send_to_all(self, messages: List[Message], node_type: NodeType):
         for _, connection in self.all_connections.items():
             if connection.connection_type is node_type:
                 for message in messages:
-                    await connection.send_message(message)
+                    if sent_message_response_ok(ProtocolMessageTypes(message.type), None):
```

Comment:
iiuc, what this check is doing is to ensure the message we send to all peers here does not expect a response. I take it this is because `send_to_all()` may only be used for such messages. That name, `send_to_all()` doesn't suggest this restriction to me. I would expect there to be a comment indicating this at the top of this function, or something.

The error you log, indicates a programming error, right? Someone sent the wrong kind of message via this function, which isn't supported. I would think we would want to do something more drastic when that happens. Maybe `assert` and make sure we disconnect, or something.

Reply:
Re: `send_to_all`:

I agree that the name doesn't suggest that restriction. I took a look at which on-the wire messages were being sent, and added this extra restriction based on what I observed.

--

Right - this is a programming logic error, BUT it is either in us, or our peer.
I did have a version of this that threw an exception, but here's why I think we shouldn't abort:

1. This depends on a mismatch between what we sent, and what the other client returned
2. Other client might be wrong, or attacking us

Reply:
I'll change this to disconnect the peer and tag you and Almog. Thanks!